index/scorch/scorch.go

```diff
@@ -37,6 +37,14 @@ const Name = "scorch"
 
 const Version uint8 = 1
 
+// UnInvertIndex is implemented by various scorch index implementations
+// to provide the un inverting of the postings or other indexed values.
+type UnInvertIndex interface {
+	// apparently need better namings here..
```

Comment:
any specific names you don't like or want suggestions for? or can we remove this comment?

Reply:
any suggestions on the interface & method name ? :) I don't like both of them.. just given some names for me to proceed..

Reply:
Here is one suggestion:

https://golang.org/doc/effective_go.html#interface-names

"By convention, one-method interfaces are named by the method name plus an -er suffix or similar modification to construct an agent noun: Reader, Writer, Formatter, CloseNotifier etc."

That doesn't quite fit in this case, because the closest to that would be `DocumentFieldTermVisitor` which is already taken to define the type for the callback itself.  Maybe we go with `DocumentFieldTermVisitable`?  I don't love that since it doesn't follow the convention, but I don't really want to rename the higher-level code at this time.  Does that work for you?